### ISSUE TITLE
ARG_VAR_MIXED not valid when assigning arguments to variables scopes

### ISSUE THREAD

**Issue by lmajano** [cflint/CFLint#540]
If you have the following code:

```java
 variables.serviceURL    = arguments.serviceURL;
variables.APIKey        = arguments.APIKey;
variables.version 		= arguments.version;
```

And you use the `ARG_VAR_MIXED` it is reporting it as bad, but it is valid since you are assigning an incoming argument to a variables scope for persistence.

**Comment by natewaite** [comment]
I am also noticing that if you have the following code
`<cffunction>
<cfargument name="status" type="any" required="yes" hint="" />
    <cfset someVar = someFunction(arguments.status)/>
</cffunction>	`

This is also triggering this rule.

**Comment by jlamoree** [comment]
I'm not sure if it's related to this issue exactly, but it could probably be fixed at the same time. This chunk of sample code triggers the `ARG_VAR_MIXED` rule because the name of the argument is `name`:
```
/**
* This is a hint.
* @name A name
*/
public string function sayHello(required string name) {
    return "Hello, " & arguments.name;
}
```
Changing the function slightly makes the check pass:
```
/**
* This is a hint.
* @nom A name
*/
public string function sayHello(required string nom) {
    return "Hello, " & arguments.nom;
}
```
It seems like `ARG_VAR_MIXED` ought not care about the name of my argument.

**Comment by ryaneberly** [comment]
think I got it.  that was a bit of a challenge.
